sources/packages/web/src/components/common/NoticeOfAssessmentFormView.vue

```diff
@@ -1,29 +1,37 @@
 <template>
-  <formio formName="noticeofassessment" :data="initialData"></formio>
+  <formio-container formName="noticeOfAssessment" :formData="initialData" />
 </template>
 
 <script lang="ts">
-import { ref, watch } from "vue";
+import { watch, defineComponent, ref } from "vue";
 import { StudentAssessmentsService } from "@/services/StudentAssessmentsService";
 import { AssessmentNOAAPIOutDTO } from "@/services/http/dto";
 
-export default {
+interface NoticeOfAssessment extends AssessmentNOAAPIOutDTO {
+  viewOnly?: boolean;
+}
+
+export default defineComponent({
   props: {
     assessmentId: {
       type: Number,
       required: true,
     },
+    viewOnly: {
+      type: Boolean,
+      required: false,
+    },
   },
-  setup(props: any) {
-    const initialData = ref({} as AssessmentNOAAPIOutDTO);
-
+  setup(props) {
+    const initialData = ref<NoticeOfAssessment>();
     watch(
-      () => props.assessmentId,
+      () => [props.assessmentId, props.viewOnly],
```

Comment:
Why do we need to call the API again if the `props.viewOnly` changes?

Reply:
it is because when the assessment is confirmed (it stays on the same page ), we need to hide the text and btns (reloading the page). ![image](https://user-images.githubusercontent.com/77353155/195954946-f4d6161c-bfe9-4ff2-897f-e786a18df0b4.png)